jerry-core/api/jerry.c

```diff
@@ -447,6 +447,26 @@ jerry_eval (const jerry_char_t *source_p, /**< source code */
 } /* jerry_eval */
 
 /**
+ * Run enqueued Promise jobs until the first thrown error or until all get executed.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return result of last executed job, may be error value.
+ */
+jerry_value_t
+jerry_run_all_enqueued_jobs (void)
```

Comment:
But those jobs are only executed in the same `jerry_run_enqueued_jobs` call if the promise can be resolved immediately, don't they? Which is not the classic use of promises. And even that only causes problems if a job always enqueues itself (or something like that) immediately. Which is really an infinite loop, analogous to `while (true) {}`, but that's a programmer error. We have no way of preventing while-true either. (Other than the recently added stop-the-vm callback, but that'd work here, too.)